### PR TITLE
refactor: the timer should not start when the user ignores the 'Todayplan' popup

### LINKED ISSUE #3156: [Bug] Planned | The issue is that Tracker start tracking time without clicking 'Start working'
The issue is when user has Plan Today, clicking 'Start' tracking the time, the system calls the popup.

The Plan Today has several tasks (some estimated, some not) but when clicks 'X' button (did not provide estimate and so on) it closes the popup and start tracking the time.

**Expect**: it should start tracking the time in two cases
1. When the user meets all requirements in the popup and click 'Start working' on the popup
2. When the user meets all requirements and closes the popup then click 'Start' tracking time on the Home page

Important: it should NOT happen in all cases

Currently
Even button 'Start working' is disabled, but it started tracking clicking on 'x'

Screenshot at Oct 17 09-02-16

### PR DESCRIPTION
**Pull request by CREDO23**
## Description

This PR fixes #3156 

## Type of Change

-   [x] Bug fix
-   [ ] New feature
-   [ ] Breaking change
-   [ ] Documentation update

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [x] My changes generate no new warnings

## Previous screenshots

[Loom](https://www.loom.com/share/6feaed8e21914ba4971d26d7074c32de?sid=0b268d6e-193f-474f-a57f-065aa5f57ef6)

## Current screenshots

[Loom](https://www.loom.com/share/2dee856380c746dea2a8cb0c4b4a6fbe?sid=81a33f92-4c03-4839-abc0-9d2ffa8845bf)
[Loom](https://www.loom.com/share/1fca46ad703c479a94e6b5f69facdc50?sid=98f804d1-3467-4d96-a595-9d8db4a7ed6e)

## Summary by CodeRabbit

- **New Features**
	- Enhanced loading indication with a spinner during submission in the daily plan modals.
	- Updated functionality for submitting planned work time in the task estimation modal.
	- Added new property to control modal behavior on outside clicks.
	- Introduced new localization strings for deleting plans across multiple languages.

- **Bug Fixes**
	- Improved error handling during submission processes.

- **Refactor**
	- Renamed and modified functions for better clarity and functionality in managing modal submissions.

### CODE REVIEW

File: apps/web/lib/features/daily-plan/add-daily-plan-work-hours-modal.tsx [hunk 1]
```diff
@@ -19,22 +19,35 @@ export function AddDailyPlanWorkHourModal(props: IAddDailyPlanWorkHoursModalProp
...
+
+	const handleSubmit = useCallback(async () => {
+		try {
+			setLoading(true);
+
+			// Update the plan work time only if the user changed it
+			plan &&
+				plan.workTimePlanned !== workTimePlanned &&
+				(await updateDailyPlan({ workTimePlanned }, plan.id ?? ''));
+
```
[the remark this answers is not in the document]
Comment (Cedric921) [reply apps/web/lib/features/daily-plan/add-daily-plan-work-hours-modal.tsx:41]: Can you use this suggestion, i think it better readable
@CREDO23

File: apps/web/lib/features/daily-plan/add-daily-plan-work-hours-modal.tsx [hunk 2]
```diff
@@ -66,19 +79,23 @@ export function AddDailyPlanWorkHourModal(props: IAddDailyPlanWorkHoursModalProp
...
 						>
 							{t('common.SKIP_ADD_LATER')}
 						</Button>
 						<Button
 							variant="default"
 							type="submit"
-							disabled={requirePlan ? (hasWorkHours ? false : true) : false}
-							className="py-3 px-5 rounded-md font-light text-md dark:text-white"
+							disabled={loading || (requirePlan ? (hasWorkHours ? false : true) : false)}
+							className="py-3 px-5 min-w-[10rem] rounded-md font-light text-md dark:text-white"
```
[the remark this answers is not in the document]
Comment (Cedric921) [reply apps/web/lib/features/daily-plan/add-daily-plan-work-hours-modal.tsx:91]: Can you use this Condition Suggestion `disabled={loading || (requirePlan && !hasWorkHours)}`
@CREDO23

File: apps/web/lib/features/daily-plan/add-task-estimation-hours-modal.tsx [hunk 3]
```diff
@@ -489,7 +501,7 @@ export function AddTasksEstimationHoursModal(props: IAddTasksEstimationHoursModa
 	return (
 		<>
 			{isRenderedInSoftFlow ? (
-				<Modal isOpen={isOpen} closeModal={closeModalAndStartTimer} showCloseIcon={requirePlan ? false : true}>
+				<Modal isOpen={isOpen} closeModal={closeModalAndSubmit} showCloseIcon={requirePlan ? false : true}>
```
[the remark this answers is not in the document]
Comment (Cedric921) [reply apps/web/lib/features/daily-plan/add-task-estimation-hours-modal.tsx:504]: Check also this Suggestion 
@CREDO23

**Review by Cedric921 — CHANGES_REQUESTED** [review]

File: apps/web/lib/features/daily-plan/suggest-daily-plan-modal.tsx [hunk 4]
```diff
@@ -14,27 +14,38 @@
...
 	const path = usePathname();
 	const t = useTranslations();
 
 	const currentDate = useMemo(() => new Date().toISOString().split('T')[0], []);
 
 	const handleCloseModal = useCallback(() => {
-		localStorage.setItem(HAS_SEEN_DAILY_PLAN_SUGGESTION_MODAL, currentDate);
+		if (!requirePlan || (requirePlan && hasPlan)) {
+			localStorage.setItem(HAS_SEEN_DAILY_PLAN_SUGGESTION_MODAL, currentDate);
+		}
```
[the remark this answers is not in the document]
Comment (evereq) [reply apps/web/lib/features/daily-plan/suggest-daily-plan-modal.tsx:33]: @CREDO23 why you not follow those!?

File: apps/web/lib/features/user-profile-plans.tsx [hunk 5]
```diff
@@ -77,6 +82,9 @@ export function UserProfilePlans() {
 	const dailyPlanSuggestionModalDate = window && window?.localStorage.getItem(DAILY_PLAN_SUGGESTION_MODAL_DATE);
 	const path = usePathname();
 	const haveSeenDailyPlanSuggestionModal = window?.localStorage.getItem(HAS_SEEN_DAILY_PLAN_SUGGESTION_MODAL);
+	const { hasPlan } = useTimer();
+	const { activeTeam } = useTeamTasks();
+	const requirePlan = useMemo(() => activeTeam?.requirePlanToTrack, [activeTeam?.requirePlanToTrack]);
```
[the remark this answers is not in the document]
Comment (evereq) [reply apps/web/lib/features/user-profile-plans.tsx:87]: @CREDO23  ^^^^

**Pull request merged**